plugins/auth-backend/src/providers/github/provider.ts

```diff
@@ -72,42 +114,112 @@ export class GithubAuthProvider implements OAuthHandlers {
-    } = await executeFrameHandlerStrategy<OAuthResult>(req, this._strategy);
+    const { result, privateInfo } = await executeFrameHandlerStrategy<
+      GithubOAuthResult,
+      PrivateInfo
+    >(req, this._strategy);
 
-    const profile = makeProfileInfo(
-      {
-        ...fullProfile,
-        id: fullProfile.username || fullProfile.id,
-        displayName:
-          fullProfile.displayName || fullProfile.username || fullProfile.id,
-      },
-      params.id_token,
+    return {
+      response: await this.handleResult(result),
+      refreshToken: privateInfo.refreshToken,
+    };
+  }
+
+  async refresh(req: OAuthRefreshRequest): Promise<OAuthResponse> {
+    const { accessToken, params } = await executeRefreshTokenStrategy(
+      this._strategy,
+      req.refreshToken,
+      req.scope,
+    );
+    const fullProfile = await executeFetchUserProfileStrategy(
+      this._strategy,
+      accessToken,
     );
+    return this.handleResult({
+      fullProfile,
+      params,
+      accessToken,
+      refreshToken: req.refreshToken,
+    });
+  }
 
-    return {
-      response: {
-        profile,
-        providerInfo: {
-          accessToken,
-          scope: params.scope,
-          expiresInSeconds: params.expires_in,
-        },
-        backstageIdentity: {
-          id: fullProfile.username || fullProfile.id,
-        },
+  private async handleResult(result: GithubOAuthResult) {
+    const { profile } = await this.authHandler(result);
+
+    const expiresInStr = result.params.expires_in;
+    const response: OAuthResponse = {
+      providerInfo: {
+        accessToken: result.accessToken,
+        scope: result.params.scope,
+        expiresInSeconds:
+          expiresInStr === undefined ? undefined : Number(expiresInStr),
       },
+      profile,
     };
+
+    if (this.signInResolver) {
+      response.backstageIdentity = await this.signInResolver(
+        {
+          result,
+          profile,
+        },
+        {
+          tokenIssuer: this.tokenIssuer,
+          catalogIdentityClient: this.catalogIdentityClient,
+          logger: this.logger,
+        },
+      );
+    }
+
+    return response;
   }
 }
 
-export type GithubProviderOptions = {};
+export const githubDefaultSignInResolver: SignInResolver<GithubOAuthResult> =
+  async (info, ctx) => {
+    const { fullProfile } = info.result;
+
+    const userId = fullProfile.username || fullProfile.id;
+
+    const token = await ctx.tokenIssuer.issueToken({
+      claims: { sub: userId, ent: [`user:default/${userId}`] },
+    });
+
+    return { id: userId, token };
+  };
+
+export type GithubProviderOptions = {
+  /**
+   * The profile transformation function used to verify and convert the auth response
+   * into the profile that will be presented to the user.
+   */
+  authHandler?: AuthHandler<GithubOAuthResult>;
+
+  /**
+   * Configure sign-in for this provider, without it the provider can not be used to sign users in.
+   */
```

Comment:
double comments

Reply:
Most of it doesn't make sense, talking about the google `'email'` thing that never got merged etc, copied to all the providers x)